app/src/main/java/com/github/libretube/api/NewPipeMediaServiceRepository.kt

```diff
@@ -319,14 +321,50 @@ class NewPipeMediaServiceRepository : MediaServiceRepository {
         )
     }
 
+    @OptIn(ExperimentalStdlibApi::class)
     override suspend fun getSegments(
         videoId: String,
-        category: String,
-        actionType: String?
-    ): SegmentData = SegmentData()
+        category: List<String>,
+        actionType: List<String>?
+    ): SegmentData {
+        // use hashed video id for privacy
+        // https://wiki.sponsor.ajay.app/w/API_Docs#GET_/api/skipSegments/:sha256HashPrefix
+        val hashedId = MessageDigest.getInstance("SHA-256")
```

Comment:
I would prefer moving the hashing function to a helper method since its implementation details are unrelated to the logic of this method, it could even make sense to create an extension function `String.sha256sum()`.